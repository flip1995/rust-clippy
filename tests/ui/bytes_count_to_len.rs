#![warn(clippy::bytes_count_to_len)]

fn main() {
    let s1 = String::from("world");

    //test warning against a string literal
    "hello".bytes().count();

    //test warning against a string variable
    s1.bytes().count();

    //make sure using count() normally doesn't trigger warning
    let vector = [0, 1, 2];
    let size = vector.iter().count();
}
