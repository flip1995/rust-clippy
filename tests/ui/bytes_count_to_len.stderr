error: using long and hard to read `.bytes().count()`
  --> $DIR/bytes_count_to_len.rs:7:5
   |
LL |     "hello".bytes().count();
   |     ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::bytes-count-to-len` implied by `-D warnings`
   = note: `.len()` achieves same functionality

error: using long and hard to read `.bytes().count()`
  --> $DIR/bytes_count_to_len.rs:10:5
   |
LL |     s1.bytes().count();
   |     ^^^^^^^^^^^^^^^^^^
   |
   = note: `.len()` achieves same functionality

error: aborting due to 2 previous errors

